pkg/scheduler/core/generic_scheduler.go

```diff
@@ -389,12 +394,21 @@ func (g *genericScheduler) getLowerPriorityNominatedPods(pod *v1.Pod, nodeName s
 
 // numFeasibleNodesToFind returns the number of feasible nodes that once found, the scheduler stops
 // its search for more feasible nodes.
-func (g *genericScheduler) numFeasibleNodesToFind(numAllNodes int32) int32 {
-	if numAllNodes < minFeasibleNodesToFind || g.percentageOfNodesToScore <= 0 ||
-		g.percentageOfNodesToScore >= 100 {
+func (g *genericScheduler) numFeasibleNodesToFind(numAllNodes int32) (numNodes int32) {
+	if numAllNodes < minFeasibleNodesToFind || g.percentageOfNodesToScore >= 100 {
 		return numAllNodes
 	}
-	numNodes := numAllNodes * g.percentageOfNodesToScore / 100
+
+	if g.percentageOfNodesToScore <= 0 {
```

Comment:
I would modify the code a bit to avoid the `else` part:

```suggestion
	adaptivePercentage := g.percentageOfNodesToScore
	if adaptivePercentage <= 0 {
		// This is a linear formula that produces 50 for a 100-node cluster and 10 for a 5000-node cluster.
		adaptivePercentage = schedulerapi.DefaultPercentageOfNodesToScore - numAllNodes/125
 		if adaptivePercentage < minFeasibleNodesPercentageToFind {
 			adaptivePercentage = minFeasibleNodesPercentageToFind
 		}
 	}
 	numNodes = numAllNodes * adaptivePercentage / 100
```

Reply:
Cool, delicate ! thanks.